src/pages/messages/components/NviCandidatePage.tsx

```diff
@@ -54,22 +49,51 @@ export const NviCandidatePage = ({ nviListQuery }: NviCandidatePageProps) => {
     meta: { errorMessage: t('feedback.error.get_registration') },
   });
 
-  const periodStatus = nviCandidate?.periodStatus.status;
+  const nviListQuery = location.state?.nviQuery;
+  const thisCandidateOffset = location.state?.currentOffset;
+
+  const hasOffset = typeof thisCandidateOffset === 'number';
+  const navigateCandidateSearchOffset = hasOffset ? Math.max(thisCandidateOffset - 1, 0) : null;
+  const isFirstCandidate = hasOffset && thisCandidateOffset === 0;
 
-  const nextCandidateQuery = useQuery({
-    queryKey: ['nextCandidate', 1, offsetNextCandidate, nviListQuery],
-    queryFn: offsetNextCandidate ? () => fetchNviCandidates(1, offsetNextCandidate, nviListQuery) : undefined,
+  const navigateCandidateQuery = useQuery({
+    enabled: hasOffset,
+    queryKey: ['navigateCandidates', 3, navigateCandidateSearchOffset, nviListQuery],
+    queryFn:
+      navigateCandidateSearchOffset !== null
+        ? () => fetchNviCandidates(3, navigateCandidateSearchOffset, nviListQuery)
+        : undefined,
     meta: { errorMessage: false },
     retry: false,
   });
 
-  const nextCandidateIdentifier = nextCandidateQuery.data?.hits[0]?.identifier;
+  const nextCandidateIdentifier = navigateCandidateQuery.isSuccess
+    ? navigateCandidateQuery.data.hits[isFirstCandidate ? 1 : 2]?.identifier
+    : null;
+  const previousCandidateIdentifier =
+    navigateCandidateQuery.isSuccess && !isFirstCandidate ? navigateCandidateQuery.data.hits[0]?.identifier : null;
 
-  const offsetNextCandidateState: CandidateOffsetState | undefined = offsetNextCandidate
-    ? {
-        offsetNextCandidate: offsetNextCandidate + 1,
-      }
-    : undefined;
+  const nextCandidateState: CandidateOffsetState | undefined = (() => {
+    if (thisCandidateOffset !== undefined && nviListQuery !== undefined) {
+      return {
+        currentOffset: thisCandidateOffset + 1,
+        nviQuery: nviListQuery,
+      };
+    } else {
+      return undefined;
+    }
+  })();
```

Comment:
heisann, her skjer det mye greier! 🤯  du kommer til å takke deg selv senere om du forstår en kodeblokk ved en rask gløtt. Det gjør ikke jeg her iaf, så da ville jeg heller foretrukket versjonen uten type faktisk 😅

that said, så skal det ikke mye til for å forenkle dette da (hvorfor wrappe det i en anonym funksjon? 🤔):
```suggestion
  const nextCandidateState: CandidateOffsetState | undefined =
    hasOffset && nviListQuery
      ? {
          currentOffset: thisCandidateOffset + 1,
          nviQuery: nviListQuery,
        }
      : undefined;
```

Reply:
Da kan vi fjerne `hasOffset`-sjekken på linje 110 og 126 også